### PR TITLE
ghc-lib-parser 9.6

### PR DESCRIPTION
**Pull request by amesgen**
- Upgrade to ghc-lib-parser-9.6 and Cabal-syntax-3.10 (see CHANGELOG entry)
 - CI: update to GHC 9.2.7 and add GHC 9.6.1

Closes #983

### CODE REVIEW

**Comment by mrkkrp** [comment]
Looks like GHC 9.6.1 has been released. Can we try to merge this now?

**Comment by 9999years** [comment]
Any progress on this? Once Ormolu cuts a release for GHC 9.6 we can work on a Fourmolu release for GHC 9.6.

**Comment by amesgen** [comment]
> Any progress on this? Once Ormolu cuts a release for GHC 9.6 we can work on a Fourmolu release for GHC 9.6.

It will still take a couple of days until someone has time to review this; but after that, a release should follow swiftly.

**Review by mrkkrp — APPROVED** [review]

File: src/Ormolu/Printer/Meat/Declaration/Value.hs [hunk 1]
```diff
@@ -591,9 +578,9 @@ p_hsExpr' s = \case
   HsVar _ name -> p_rdrName name
   HsUnboundVar _ occ -> atom occ
   HsRecSel _ fldOcc -> p_fieldOcc fldOcc
-  HsOverLabel _ v -> do
+  HsOverLabel _ sourceText _ -> do
     txt "#"
-    atom v
+    p_sourceText sourceText
```
[the remark this answers is not in the document]
Comment (mrkkrp) [reply src/Ormolu/Printer/Meat/Declaration/Value.hs:583]: Feel free a follow-up PR for this, but I think it is really optional.

File: src/Ormolu/Parser.hs [hunk 2]
```diff
@@ -108,15 +110,19 @@ parseModuleSnippet Config {..} fixityMap dynFlags path rawInput = liftIO $ do
...
-            showErr =
-              showOutputable
-                . formatBulleted defaultSDocContext
-                . diagnosticMessage
-                . errMsgDiagnostic
+            showErr (errMsgDiagnostic -> err) = codeMsg <> msg
+              where
+                codeMsg = case diagnosticCode err of
+                  Just code -> "[" <> showOutputable code <> "] "
+                  Nothing -> ""
```
[the remark this answers is not in the document]
Comment (mrkkrp) [reply src/Ormolu/Parser.hs:117]: The same here, seems rather optional, but feel free to open a follow-up PR.

**Pull request merged**